src/main/java/core/basesyntax/JavaStreamApi.java

```diff
@@ -1,90 +1,123 @@
+                .count();
     }
 
     /**
-     * <p>3. Дана коллекция строк List of String elements
+     * 3. Дана коллекция строк List of String elements
      * (пример: Arrays.asList(«a1», «a2», «a3», «a1»)).
-     * Вернуть Optional первого элемента коллекции</p>
+     * Вернуть Optional первого элемента коллекции
      **/
     public Optional<String> firstElement(List<String> elements) {
-        return null;
+        return elements.stream()
+                .findFirst();
     }
 
     /**
-     * <p>4. Дана коллекция строк List of String elements
+     * 4. Дана коллекция строк List of String elements
      * (приметр: Arrays.asList(«a1», «a2», «a3», «a1»)).
-     * Найти элемент в коллекции равный `element` или кинуть ошибку NoSuchElementException</p>
+     * Найти элемент в коллекции равный `element` или кинуть ошибку NoSuchElementException
      **/
     public String findElement(List<String> elements, String element) {
-        return null;
+        return elements.stream()
+                .filter(listElement -> listElement.equals(element))
+                .findAny()
+                .orElseThrow();
     }
 
     /**
-     * <p>5. Дана коллекция чисел List of Integer numbers (пример: Arrays.asList(6, 2, 3, 7, 2, 5))
+     * 5. Дана коллекция чисел List of Integer numbers (пример: Arrays.asList(6, 2, 3, 7, 2, 5))
      * Отнимите от каждого элемента, который стоит на непарной позиции (имеет не парный индекс) 1
      * и верните среднее арифметическое всех нечетных чисел или киньте ошибку
-     * NoSuchElementException</p>
+     * NoSuchElementException
      **/
     public Double averageSumOdd(List<Integer> numbers) {
-        return null;
+        iterate(1, index -> index + 2)
+                .limit(numbers.size() / 2)
+                .forEach(index -> numbers.set(index, numbers.get(index) - 1));
+        return numbers.stream()
+                .filter(number -> number % 2 != 0)
+                .mapToDouble(Integer::doubleValue)
+                .average()
+                .orElseThrow();
     }
 
     /**
-     * <p>6. Дана коллекция класс People (с полями name — имя, age — возраст, sex — пол),
+     * 6. Дана коллекция класс People (с полями name — имя, age — возраст, sex — пол),
      * вида Arrays.asList( new People(«Вася», 16, Sex.MAN),
      * new People(«Петя», 23, Sex.MAN),
-     * new People(«Елена», 42, Sex.WOMEN),
+     * new People(«Елена», 42, Sex.WOMAN),
      * new People(«Иван Иванович», 69, Sex.MAN)).
-     * Задача: Выбрать мужчин-военнообязанных (от `fromAge` до `toAge` лет)</p>
+     * Задача: Выбрать мужчин-военнообязанных (от `fromAge` до `toAge` лет)
      **/
     public List<People> manSelectByAge(List<People> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(person -> person.getSex().equals(People.Sex.MAN)
+                        && person.getAge() >= fromAge
```

Comment:
@RomanMinevich Can you replace this with several filters?

Reply:
Done it.